reactor-core/src/test/java/reactor/core/publisher/FluxReplayTest.java

```diff
@@ -303,7 +306,7 @@ protected void hookOnNext(Integer value) {
 		replay.subscribe(threeThenEightSubscriber);
 		replay.connect();
 
-		assertThat(requests).containsExactly(5L);
+		assertThat(requests).containsExactly(5L, 8L);
```

Comment:
Test change alert:
was it a wrong assumption that was supposed to catch the issue or we change the behaviour of the old cases?

Reply:
actually this made me find an issue in the previous implementation, updated the whole thing